### PR TITLE
Added OSE support and documentation

### PR DESCRIPTION
**Pull request by sabre1041**
Added support for building and running the builder and demo in OpenShift

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -15,6 +17,11 @@ LABEL io.k8s.description="ASP.NET 5.0" \
 
 RUN printf "deb http://ftp.us.debian.org/debian jessie main\n" >> /etc/apt/sources.list
```
Comment (jawnsy) [inline Dockerfile]: @sabre1041 Thanks for using a single `RUN` instruction. Can you combine the others as well?

@alberttwong FYI, using multiple RUN instructions results in Docker creating extra layers, so it's better practice to use a multi-line RUN statement as Andrew has done for his changes.

File: Dockerfile [hunk 2]
```diff
@@ -15,6 +17,11 @@ LABEL io.k8s.description="ASP.NET 5.0" \
...
 RUN apt-get -qq update && apt-get install -qqy sqlite3 libsqlite3-dev && rm -rf /var/lib/apt/lists/*
+RUN mkdir -p ${HOME} && \
+    chown -R 1001:0 ${HOME}/ && \
+    useradd -u 1001 -r -g 0 -d ${HOME} -s /sbin/nologin -c "Default Application User" default && \
+    chown -R 1001:0 /opt/app-root
+
```
Comment (jawnsy) [inline Dockerfile]: @sabre1041 Please remove the extra blank line

File: README.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ High level process
 3. Run the new OpenShift ASP.NET base image
 
 For more information about using these images with OpenShift, please see
-the official OpenShift Documentation (https://docs.openshift.org/latest/using_images/s2i_images/php.html) .
+the official OpenShift Documentation (https://docs.openshift.com/enterprise/3.1/creating_images/s2i.html) .
```
Comment (jawnsy) [inline README.md:16]: @sabre1041 Is it better to link to OSE or the Origin docs? hmm... The community docs might be more up-to-date https://docs.openshift.org/latest/creating_images/s2i.html

File: Dockerfile (hunk 1 above)
Comment (alberttwong) [inline Dockerfile]: @jawnsy I debated about this but opted to follow MS' example.   I'm fine to collapse.

File: README.md (hunk 3 above)
Comment (sabre1041) [inline README.md:16]: @jawnsy The origin docs will have the most up to date information but may contain features that are not available in the enterprise version. I tend to prefer referencing enterprise docs

File: Dockerfile (hunk 1 above)
Comment (sabre1041) [inline Dockerfile]: @alberttwong @jawnsy do you want me to collapse it all in this PR or fix my changes, merge and then combine?

**Comment by sabre1041** [comment]
@jawnsy updated PR with suggested changes

**Pull request merged**